cmd/auth/main.go

```diff
@@ -20,217 +15,142 @@ import (
+	KetoWritePort string        `env:"MF_KETO_WRITE_REMOTE_PORT"     envDefault:"4467"`
+	LoginDuration time.Duration `env:"MF_AUTH_LOGIN_TOKEN_DURATION"  envDefault:"10h"`
+	JaegerURL     string        `env:"MF_JAEGER_URL"                 envDefault:"localhost:6831"`
 }
 
 func main() {
-	cfg := loadConfig()
 	ctx, cancel := context.WithCancel(context.Background())
 	g, ctx := errgroup.WithContext(ctx)
-	logger, err := logger.New(os.Stdout, cfg.logLevel)
-	if err != nil {
-		log.Fatalf(err.Error())
-	}
-
-	db := connectToDB(cfg.dbConfig, logger)
-	defer db.Close()
-
-	tracer, closer := initJaeger("auth", cfg.jaegerURL, logger)
-	defer closer.Close()
 
-	dbTracer, dbCloser := initJaeger("auth_db", cfg.jaegerURL, logger)
-	defer dbCloser.Close()
-
-	readerConn, writerConn := initKeto(cfg.ketoReadHost, cfg.ketoReadPort, cfg.ketoWriteHost, cfg.ketoWritePort, logger)
-
-	svc := newService(db, dbTracer, cfg.secret, logger, readerConn, writerConn, cfg.loginDuration)
-
-	g.Go(func() error {
-		return startHTTPServer(ctx, tracer, svc, cfg.httpPort, cfg.serverCert, cfg.serverKey, logger)
-	})
-	g.Go(func() error {
-		return startGRPCServer(ctx, tracer, svc, cfg.grpcPort, cfg.serverCert, cfg.serverKey, logger)
-	})
-
-	g.Go(func() error {
-		if sig := errors.SignalHandler(ctx); sig != nil {
-			cancel()
-			logger.Info(fmt.Sprintf("Authentication service shutdown by signal: %s", sig))
-		}
-		return nil
-	})
-	if err := g.Wait(); err != nil {
-		logger.Error(fmt.Sprintf("Authentication service terminated: %s", err))
+	// create auth service configurations
+	cfg := config{}
+	// load auth service configurations from environment
+	if err := env.Parse(&cfg); err != nil {
+		log.Fatalf("failed to load %s configuration : %s", svcName, err.Error())
 	}
-}
 
-func loadConfig() config {
-	dbConfig := postgres.Config{
-		Host:        mainflux.Env(envDBHost, defDBHost),
-		Port:        mainflux.Env(envDBPort, defDBPort),
-		User:        mainflux.Env(envDBUser, defDBUser),
-		Pass:        mainflux.Env(envDBPass, defDBPass),
-		Name:        mainflux.Env(envDB, defDB),
-		SSLMode:     mainflux.Env(envDBSSLMode, defDBSSLMode),
-		SSLCert:     mainflux.Env(envDBSSLCert, defDBSSLCert),
-		SSLKey:      mainflux.Env(envDBSSLKey, defDBSSLKey),
-		SSLRootCert: mainflux.Env(envDBSSLRootCert, defDBSSLRootCert),
+	// create new logger
+	logger, err := logger.New(os.Stdout, cfg.LogLevel)
+	if err != nil {
+		log.Fatalf(err.Error())
 	}
 
-	loginDuration, err := time.ParseDuration(mainflux.Env(envLoginDuration, defLoginDuration))
+	// Postgres client
+	// Create postgres client configuration with default values for database
+	dbConfig := pgClient.Config{Name: defDB}
+	// create new postgres client
+	db, err := pgClient.SetupWithDefConfig(envPrefix, *authPg.Migration(), dbConfig)
 	if err != nil {
-		log.Fatal(err)
+		log.Fatalf("failed to setup postgres database : %s", err.Error())
 	}
+	defer db.Close()
```

Comment:
No need for these fine-grained comments. One comment on this entire block, like  `Setup connection to DB`, is fine.